src/common/variables.c

```diff
@@ -187,71 +190,144 @@ static void init_expansion(dt_variables_params_t *params, gboolean iterate)
-  else if(has_prefix(variable, "MSEC"))
-    result = g_strdup_printf("%.3d", params->data->time.msec);
-
-  else if(has_prefix(variable, "EXIF_YEAR"))
-    result = g_strdup_printf("%.4d", datetime.year);
-  else if(has_prefix(variable, "EXIF_MONTH"))
-    result = g_strdup_printf("%.2d", datetime.month);
-  else if(has_prefix(variable, "EXIF_DAY"))
-    result = g_strdup_printf("%.2d", datetime.day);
-  else if(has_prefix(variable, "EXIF_HOUR"))
-    result = g_strdup_printf("%.2d", datetime.hour);
-  else if(has_prefix(variable, "EXIF_MINUTE"))
-    result = g_strdup_printf("%.2d", datetime.minute);
-  else if(has_prefix(variable, "EXIF_SECOND"))
-    result = g_strdup_printf("%.2d", datetime.second);
-  else if(has_prefix(variable, "EXIF_MSEC"))
-    result = g_strdup_printf("%.3d", datetime.msec);
-  else if(has_prefix(variable, "EXIF_ISO"))
+  char exif_datetime[DT_DATETIME_LENGTH];
+  GDateTime *datetime = params->data->have_exif_dt ? params->data->datetime : params->data->time;
+
+  if(_has_prefix(variable, "YEAR") || _has_prefix(variable, "DATE.LONG_YEAR"))
+    result = g_date_time_format(params->data->time, "%Y");
+  else if(_has_prefix(variable, "SHORT_YEAR") || _has_prefix(variable, "DATE.SHORT_YEAR"))
+    result = g_date_time_format(params->data->time, "%y");
+  else if(_has_prefix(variable, "MONTH") || _has_prefix(variable, "DATE.MONTH"))
+    result = g_date_time_format(params->data->time, "%m");
+  else if(_has_prefix(variable, "SHORT_MONTH") || _has_prefix(variable, "DATE.SHORT_MONTH"))
+    result = g_date_time_format(params->data->time, "%b");
+  else if(_has_prefix(variable, "LONG_MONTH") || _has_prefix(variable, "DATE.LONG_MONTH"))
+    result = g_date_time_format(params->data->time, "%B");
+  else if(_has_prefix(variable, "DAY") || _has_prefix(variable, "DATE.DAY"))
+    result = g_date_time_format(params->data->time, "%d");
+  else if(_has_prefix(variable, "HOUR_AMPM") || _has_prefix(variable, "DATE.HOUR_AMPM"))
+    result = g_date_time_format(params->data->time, "%I %p");
+  else if(_has_prefix(variable, "HOUR") || _has_prefix(variable, "DATE.HOUR"))
+    result = g_date_time_format(params->data->time, "%H");
+  else if(_has_prefix(variable, "MINUTE") || _has_prefix(variable, "DATE.MINUTE"))
+    result = g_date_time_format(params->data->time, "%M");
+  else if(_has_prefix(variable, "SECOND") || _has_prefix(variable, "DATE.SECOND"))
+    result = g_date_time_format(params->data->time, "%S");
+  else if(_has_prefix(variable, "MSEC"))
+  {
+    result = g_date_time_format(params->data->time, "%f");
+    result[3] = '\0';
+  }
+  // for watermark backward compatibility
+  else if(_has_prefix(variable, "DATE"))
+  {
+    dt_datetime_gdatetime_to_exif(exif_datetime, params->data->show_msec ? DT_DATETIME_LENGTH : DT_DATETIME_EXIF_LENGTH, params->data->time);
+    result = g_strdup(exif_datetime);
+  }
+
+  else if(_has_prefix(variable, "EXIF_YEAR") || _has_prefix(variable, "EXIF.DATE.LONG_YEAR"))
```

Comment:
Likewise here `LONG_YEAR` better to be `YEAR.LONG`, and reads better to me `EXIF.DATE.YEAR.LONG`.